docs: add high level overview (#435)
kubernetes-sigs/blixt: docs/high_level_overview.md

# Design Overview
```mermaid
erDiagram
    k8sApi {
        GatewayApi CRDs
    }
    k8sCluster one to one k8sApi : has
    k8sCluster one to one or many Nodes : "has"
    k8sCluster one to one or many Services : "runs"
    Nodes one to one Dataplane : "runs"
    
    Services one to one or many Controlplane : "loadbalances"
    Controlplane one to one or many Nodes : "runs on"

    Dataplane {
        Daemonset Pod
        Blixt-Binary loader
        Elevated Privileges
        eBPF Kernel
    }
    Kernel {
        eBPF Networking
    }
    

>     Dataplane one to one Kernel : "controls via syscalls"

Reviewer: ```suggestion
    Dataplane one to one Kernel : "controls via kernel memory and bpf helpers"
```

    
    Controlplane {
        Deployment Pod
        Blixt-Binary controller
        k8s Controllers
    }
    
    Controlplane one to one or many Dataplane : "controls via gRPC"

>     Controlplane one to one or many k8sApi : "interacts with"

Reviewer: ```suggestion
    Controlplane one to one k8sApi : "is programmed by"
```

Author: Updated.

    
    Kernel one to one Nodes : "host networking"
```

# GatewayApi Config Flow

```mermaid
flowchart LR
    User[/User/]
    k8sResources(["k8s Resources\nGatewayApi CRDs"])
    Controlplane("Blixt Controlplane\nk8s Service")
    Dataplane("Blixt Dataplane\nk8s Daemonset")

>     Kernel("Kernel\neBPF Networking")

Author: Interesting, the preview in the IDE was rendering it.
